app/javascript/admin/controllers/config_controller.js

```diff
@@ -265,4 +285,63 @@ export default class ConfigController extends Controller {
     }
   }
   // AUTH PROVIDERS FUNCTIONS END
+
+  enabledProvidersWithMissingKeys() {
+    const providersWithMissingKeys = [];
+    document
+      .querySelectorAll('[data-enable-auth="true"]')
+      .forEach((provider) => {
+        const providerName = provider.dataset;
+        const providerOfficialName = provider.dataset;
+        if (
+          !document.getElementById(`site_config_${providerName}_key`).value ||
+          !document.getElementById(`site_config_${providerName}_secret`).value
+        ) {
+          providersWithMissingKeys.push(providerOfficialName);
+        }
+      });
+
+    return providersWithMissingKeys;
+  }
+
+  generateProvidersList(providers) {
+    const list = providers.reduce((html, provider) => {
+      return `${html}<li>${provider}</li>`;
+    }, '');
+
+    return list;
+  }
+
+  missingAuthKeysModalBody(providers) {
+    return `
+      <p>You haven't filled out all of the required fields to  enable the following authentication providers:</p>
+      <ul class="mb-0">${this.generateProvidersList(providers)}</ul>
+      <p>If you save your configuration now, these authorization providers will not be enabled.</p>
```

Comment:
```suggestion
      <p class="mb-0">If you save your configuration now, these authorization providers will not be enabled.</p>
```

Let's add a `margin-bottom: 0` styling to this sentence so that the spacing between this text and the buttons below are not too wide.